src/app/cras/edit-cra/edit-cra.component.spec.ts

```diff
@@ -73,70 +81,216 @@ describe('EditCraComponent', () => {
+        expect(component.cra).toEqual(new Cra());
+      });
     });
-  }));
 
-  it('should switch mode when review QueryParams provided', async(() => {
-    testData.mode = 'review';
-    const spy = spyOn(serializer, 'deserialize')
-      .and.callFake(
-        (data) => {
-          expect(data).toBe(testReviewToken);
-        }
-      )
-      .and.returnValue(testData);
-    route.queryParams = of({ data: testReviewToken });
-    fixture.detectChanges();
-    component.ngOnInit();
-    fixture.whenStable().then(() => {
-      expect(component.mode).toBe(testData.mode);
-      expect(component.cra).toBe(testData.cra);
+    describe('when called in edit/review mode', () => {
+      it('should init filled with token data', async(() => {
+        testData.mode = 'edit';
+        route.queryParams = of({ data: testEditToken });
+
+        const spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        fixture.whenStable().then(() => {
+          expect(component.cra).toBe(testData.cra);
+          expect(component.consultantNameInput.value).toBe(testData.cra.consultant.name);
+          expect(component.consultantEmailInput.value).toBe(testData.cra.consultant.email);
+          expect(component.missionTitleInput.value).toBe(testData.cra.mission.title);
+          expect(component.missionFinalClientInput.value).toBe(testData.cra.mission.client);
+          expect(spyDeserialize).toHaveBeenCalledWith(testEditToken);
+        });
+      }));
+    });
+
+    describe('when called in edit mode', () => {
+      it('should set mode to edit', () => {
+        testData.mode = 'edit';
+        route.queryParams = of({ data: testEditToken });
+
+        const spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        expect(component.mode).toBe('edit');
+        expect(spyDeserialize).toHaveBeenCalledWith(testEditToken);
+      });
+
+      it('should set Acrabadabra - Editer un compte rendu d\'activité as page title', () => {
+        testData.mode = 'edit';
+        route.queryParams = of({ data: testEditToken });
+
+        const spySetTitle = spyOn(titleService, 'setTitle'),
+              spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        expect(spySetTitle).toHaveBeenCalledWith('Acrabadabra - Editer un compte rendu d\'activité');
+        expect(spyDeserialize).toHaveBeenCalledWith(testEditToken);
+      });
     });
-  }));
 
-  it('should enable modal and saved text on save', () => {
-    component.cra = testData.cra;
-    component.onSubmitCRA();
-    expect(component.saved).toBe(true);
-    expect(component.showModal).toBe(true);
+    describe('when called in review mode', () => {
```

Comment:
same here, all initialization line, mode, queryParams, spyDeserialize, should be put in a `beforeEach` to avoid duplicated code